client/src/modules/cost_center/allocation_bases/allocation_bases.js

```diff
@@ -0,0 +1,107 @@
+angular.module('bhima.controllers')
+  .controller('CostCenterAllocationBasesController', CostCenterAllocationBasesController);
+
+CostCenterAllocationBasesController.$inject = [
+  'CostCenterService',
+  'uiGridConstants',
+  'NotifyService',
+];
+
+function CostCenterAllocationBasesController(
+  CostCenters,
+  uiGridConstants,
+  Notify,
+) {
+  const vm = this;
+
+  // global variables
+  vm.gridApi = {};
+  vm.filterEnabled = false;
+
+  // options for the UI grid
+  vm.gridOptions = {
+    appScopeProvider  : vm,
+    enableColumnMenus : false,
+    showColumnFooter  : true,
+    fastWatch         : true,
+    flatEntityAccess  : true,
+    enableSorting     : false,
+    onRegisterApi     : onRegisterApiFn,
+  };
+
+  function onRegisterApiFn(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  function load() {
+    CostCenters.getAllocationKeys()
+      .then(rows => {
+        if (!rows || (rows && !rows.costCenterIndexes.length)) { return; }
+
+        const columnDefs = [{
+          field : 'label',
+          displayName : 'COST_CENTER.TITLE',
+          headerCellFilter : 'translate',
+          cellFilter : 'translate',
+          aggregationHideLabel : true,
+          aggregationType : uiGridConstants.aggregationTypes.count,
+        }];
+
+        rows.costCenterIndexes.forEach((item, idx) => {
+          if (idx >= 0) {
+            const column = {
+              field : `idx${idx}`,
+              displayName : item.index,
+              type : 'number',
+              headerCellFilter : 'translate',
+              cellClass : 'text-right',
+              footerCellClass : 'text-right',
+              cellFilter : 'number: 2',
+              footerCellFilter : 'number:2',
+              aggregationHideLabel : true,
+              aggregationType : uiGridConstants.aggregationTypes.sum,
+            };
+            columnDefs.push(column);
```

Comment:
It looks like you are doing an array.forEach() and an array.push() inside it.  Could you just do an array.map()?